### ISSUE TITLE
WorkZoneRoadEventFeatureBuilder.WithLane() Does Not Behave as Expected

### ISSUE THREAD

**Issue by thenadz** [WSDOT/wzdx.models#3]
As written, this function invokes `CreateWith()` which in turn returns a whole new instance of `WorkZoneRoadEventFeatureBuilder` rather than adding provided lane to existing instance and returning `this`.

**Comment by thenadz** [comment]
I started investigating a fix for this, but wasn't quite clear what intent with CreateWith was, so paused for now. If I can get some clarity on intent I can see if I can get a PR together to address. Thanks!

**Comment by rhuebner76** [comment]
As the project was born and evolved in a closed environment, I appreciate you taking the time to provide input, so any opinionated implementations in the repository are certainly open for change. Certainly, I expect more issues and design changes to arise as it gets used more.

To help clarify, originally, these builders were meant to be immutable. Although, while originally needed, that need for immutability may be less needed now then earlier. 

While I am unsure what behavior is unexpected, it does appear there is an issue in the Lane builder adding restrictions as an Action delegate parameter was an incorrect holdover from a prior implementation.

**Comment by thenadz** [comment]
Hi Russ -

Thanks for this. In terms of what unexpected means in this case, simply lanes are not created in serializer when lane builder is used. Below is a stripped down version of how the builder is being used to try and add lanes to a feature builder.

```
wzFeatureBuilder.WithLane(
    LaneType.General,
    LaneStatus.Open,
    1,
    laneBuilder => { });
```

**Comment by rhuebner76** [comment]
Since the builders are currently immutable, **WithLane** or any of the builder methods do not directly modify the instance state, so results of the methods **must** be assigned to a variable to capture the change.

``` CS
builder = builder.WithLane( /* */ );
```

I have updated main with [Pure] attributes to the immutable builder methods. I am reviewing the need for them to be immutable.

**Comment by thenadz** [comment]
Thanks! I think this immutable approach is root of most of the issues we've been encountering. Would be very interested to hear result of this review - immutable approach you describe adds some significant complexity for less trivial implimentations where the bulider is being passed around in order to fully populate the feed. Happy to jump on a call at some point if useful to discuss our use cases and how we can best reconcile with use cases for which this library was originally designed.

**Comment by rhuebner76** [comment]
I have an update with the builders refactored and updated to remove the immutability, yet before I push it up I will be refactoring the none-nullable properties to allow for null. I am in my personal opinion, nulls are a pain and should be avoided, but the specification allows for incomplete objects of each class.

**Comment by thenadz** [comment]
Thanks, Russ! Appreciate you and the team taking the time to look at this.

**Comment by thenadz** [comment]
Confirmed fix with 0.0.6 release. Thanks!
